### ISSUE TITLE
Put the standard on GitHub

### ISSUE THREAD

**Issue by jacobwilliams** [j3-fortran/fortran_proposals#48]
Put the LaTeX code for the standard on GitHub, and allow the community to help edit it.

**Comment by certik** [comment]
I would very much like that! However the ISO doesn't allow that. Ultimately I would like to figure out some compromise. Right now we can't do it, and I agree we need to work on fixing it. How does C++ do it? They have the same problem.

**Comment by jacobwilliams** [comment]
There has to be a way since C++ does do it: https://github.com/cplusplus/draft

**Comment by certik** [comment]
Great! I was told this cannot be done. Yet C++ is doing it. Let's do what they do.

**Comment by FortranFan** [comment]
@jacobwilliams wrote:

> Put the LaTeX code for the standard on GitHub, and allow the community to help edit it.

Using an online collaborative platform - GitHub being ideal at present - for language development which then includes whatever postscript is used for the standard document is exactly what Fortran needs to rise back as the _**lingua franca**_ of scientific and technical/numerical programming.

Alas this has been discussed elsewhere in the past and this idea faces great resistance.

Ideally the core development of the Fortran language will, in a deep, philosophical sense, be independent and separate from the Fortran standard and the IEC ISO document for the Fortran standard would simply fall naturally out of the development process, as if a byproduct.  And most of the "heavylifting" to pull together the IEC ISO document would be mechanical that can be scripted/automated or otherwise be collated and refined by many sharp minds looking at it together i.e.,, the online collaborative model feasible using GitHub should work well for this.

Unfortunately though, it appears as if the standard document is the cart that is placed before the horse for quite a while now and it is the product that dictates what Fortran can be or not i.e., an inverted development model is influencing matters.  The advancement of the language likely suffers as a consequence of this.

**Comment by certik** [comment]
I agree the cart is currently before the horse. It will take time before we can get the horse at front.

In the meantime, even with the cart in front of the horse, we still need to be using github for the standard and collaborate on it. All pull requests (PRs) can still be merged just by one person initially. Until more people gain the trust of the community that they can have merge rights.

Also, this standard editing process should really be a byproduct of which proposals the committee approves. The committee itself can (almost) work like today, with formal proposals and such, as long as the committee will consider proposals that we will submit for all subsequent meetings as a result of this github collaboration.

**Comment by rweed** [comment]
At a minimum, this site should provide either a link to the F2018 interpretation document which is probably very close to the public standard (currently at https://j3-fortran.org/doc/year/18/18-007r1.pdf) or make the pdf available for download directly from this site

**Comment by certik** [comment]
@rweed thanks! I think this is what we would do if we lost the sources to the standard. But the committee has the sources (or to be more precise, at least one person on the committee has the sources), so I will try to convince the committee to put the sources on github.

**Comment by FortranFan** [comment]
@rweed wrote:

> I have converted the interpretation document (18-007r1.pdf) to HTML, TeX, and Markdown ..

I'll strongly recommend contacting WG5 and/or PL22.3 and confirm there are no copyright or other restrictions with document 18-007r1, especially with distributing any changes online anywhere even if they are in the electronic format only.

**Comment by rweed** [comment]
@FortranFan, I agree. However, there appears to be no restriction on downloading the PDF from J3 (which implies to me that redistribution is O.K.) and as I understand it WG5 only limits distribution of the final version of the standard that goes to the publisher.  Still, you are correct we should get a ruling on if there are any unstated restrictions on distribution/redistribution.
